File: backend/src/models/Label.ts
export interface ILabel {
  id: number;
  name: string;
  colour: string;
}

export class Label implements ILabel {
  id: number;
  name: string;
  colour: string;

  constructor(id: number, name: string, colour: string) {
    this.id = id;
    this.name = name;
    this.colour = colour;
  }
}

File: backend/src/models/label.py
from pydantic import BaseModel
from sqlalchemy import Column, String, UUID
from ..database import Base
import uuid

# Database model
class LabelDB(Base):
    __tablename__ = "label"
    
    id = Column(UUID, primary_key=True, default=uuid.uuid4)
    name = Column(String(255), nullable=False)
    colour = Column(String(255), nullable=False)

# API model
class Label(BaseModel):
    id: str
    name: str
    colour: str